File: problems/finished/179/std.py
from Queue import PriorityQueue

n = input()
Q = PriorityQueue()
Q.put(-input())
ans = 0
for i in range(1, n):
    x = -input()
    y = Q.get()
    if -y > -x:
        ans += -y + x
        y = x
    Q.put(x)
    Q.put(y)
print(ans)

File: problems/finished/179/std.cpp
#include<bits/stdc++.h>
using namespace std;

priority_queue<int> Q;

int main(){
    int n, x;
    cin >> n >> x;
    Q.push(x);
    long long ans = 0;
    for(int i = 2; i <= n; i++){
        scanf("%d", &x);
        if(Q.top() > x){
            ans += Q.top() - x;
            Q.pop();
            Q.push(x);
        }
        Q.push(x);
    }
    cout << ans << endl;
}
